### ISSUE TITLE
DynamoDb error when querying an object that is modelled to belong to an optional parent object

### ISSUE THREAD

**Issue by hisham** [aws-amplify/amplify-cli#262]
**Describe the bug**
If you create a model without a parent model, then queries on that model will fail. i.e. If you create a post without a parent blog, then the resolver for blog.post will fail instead of returning null. 

**To Reproduce**
Let's say I have the following schema:

```
type Blog @model {
  id: ID!
  name: String!
  post: [Post] @connection(name: "BlogPosts")
}
type Post @model {
  id: ID!
  title: String!
  blog: Blog @connection(name: "BlogPosts")
}
```

And I create a post but without a parent blog. When I do a getPost query, that query will fail when it queries the post's blog (post.blog). It will fail with the following error:  

`The provided key element does not match the schema (Service: AmazonDynamoDBv2; Status Code: 400; Error Code: ValidationException; Request ID: RV4SLCGFF2GCNH98OUNV1DGQF3VV4KQNSO5AEMVJF66Q9ASUAAJG)`

This is because the resolver on the 'blog' attribute of 'Post' is something like below. It assumes there is a $ctx.source.postBlogId, and so null is set to "id" and dynamo does not like that:

```
{
  "version": "2017-02-28",
  "operation": "GetItem",
  "key": {
      "id": $util.dynamodb.toDynamoDBJson($ctx.source.postBlogId)
  }
}
```

**Expected behavior**
blog.post should return null if it has no parent post. To accomplish this, resolver needs to be something like the following:

```
{
  "version": "2017-02-28",
  "operation": "GetItem",
  "key": {
#if( $ctx.source.postBlogId )  
      "id": $util.dynamodb.toDynamoDBJson($ctx.source.postBlogId)
#else
## Todo: figure out how to cancel ddb operation rather than do this dummy query
	  "id": $util.dynamodb.toDynamoDBJson("nothing")
#end
  }
}
```

**Comment by mmichelli** [comment]
This is driving me crazy!

**Comment by mikeparisstuff** [comment]
Update. I am tracking this and will work on getting a fix in the next few days.

**Comment by mikeparisstuff** [comment]
This is fixed in #359

**Comment by kaustavghosh06** [comment]
This issue is fixed and would be a part of the next CLI version (0.1.30).
